Simplify ToricBlowupMorphism

paemurru opened the pull request:
The previous properties `center_data` and `center_unnormalized` are not necessary anymore, since they can be easily constructed from `index_of_exceptional_ray(phi)` and `minimal_supercone_coordinates_of_exceptional_ray(phi)`.

Extra comment: previously `blow_up` checked whether the codomain is smooth using `is_smooth`. As a side effect, this set the attribute `is_smooth` for the variety if it was not set before. In turn, this changed the printing from `Normal toric variety` to `Normal, smooth toric variety`.

paemurru committed:
Simplify ToricBlowupMorphism

The previous properties `center_data` and `center_unnormalized` are not necessary anymore, since they can be easily constructed from `index_of_exceptional_ray(phi)` and `minimal_supercone_coordinates_of_exceptional_ray(phi)`.

paemurru committed:
Fix doctest

HereAround commented:
@paemurru , @HechtiDerLachs and myself just discussed this PR at length over a phone call. Here is our take:
* Keep `center_unnormalized`,
* Do not export `ToricBlowupMorphism`.

Other than that, the PR looks fine.

@paemurru will adjust accordingly, then review once more.

paemurru committed:
Fix center, unexport type

paemurru commented:
### Updates

* Renamed `center_unnormalized` to `center`.
* Changed the implementation so that it now returns the image of the exceptional prime divisor.
* Added a fast implementation for the future when ideal sheaves on nonsmooth toric varieties get implemented.

### Comments

Previously, the function `center_unnormalized` actually only worked correctly according to its documentation in the case the codomain was smooth.

Some comments on the interface in `src/AlgebraicGeometry/Schemes/BlowupMorphism/BlowupMorphism.jl`:
* The interface asks that `exceptional_divisor` is a Cartier divisor, while for weighted blowups and toric blowups, the most natural "exceptional divisor" is a prime exceptional divisor.
* The interface asks that `center(f::AbsSimpleBlowupMorphism)` is an ideal sheaf such that the blowup along it coincides with `f`. This is not the case for the `center` here, as it only returns a subvariety with the reduced induced structure.

paemurru committed:
Fix doctest

HereAround reviewed (approved):
Thank you for working on this @paemurru . Looks good to me.

Pinging @HechtiDerLachs to have another look at this.

HereAround commented:
(One required job failed. To me, this failure appears unrelated to the changes in this PR, so I just restarted the failed jobs.)

HereAround commented:
Since I have not heard from @HechtiDerLachs , I assume that this PR is fine and will merge now.

(Otherwise, can of course be improved in another PR.)
